### ISSUE TITLE
H264 streaming for iPhone

### ISSUE THREAD

**Issue by krishongithub** [mbebenita/Broadway#156]
I am working on a project where iPhone will stream H264 data over web sockets hosting a server on its end, and trying to send the stream to a web client, to draw it on a canvas, this needs a player and decoder at the web client side. I am currently using a project created for Android for the same, https://github.com/omerjerk/RemoteDroid_web, I cannot verify whether it is working on Android or not, as I am working on iPhone client. 

The Broadway description says, the file has to be downloaded completely before the player begins, but the "RemoteDroid_web" project claims, it is built for Android for H264 streaming. Will the code work if I tweak it for iPhone, or in other way, can Broadway be used to stream the H264 data and play than download-the-file-completely-and-then-play?

**Comment by soliton4** [comment]
actually broadway works excellent with a raw h264 stream.
make sure you are sticking to the base profile.
if you need higher profiles you may have to compile another decoder with emscripten and it might be considerably slower

**Comment by mbebenita** [comment]
> The Broadway description says, the file has to be downloaded completely before the player begins,

That's just a limitation of the demo. The decoder itself can be wired up to decode frames as they are downloaded.

**Comment by krishongithub** [comment]
Thanks for the quick replay, based on the information you provided, I started implementing it. But I have issues using "RawRenderer", can you please help me here. I tried it from both Safari and Chrome. Please find the attachments. The source code is at https://github.com/omerjerk/RemoteDroid_web
<img width="1440" alt="screen shot 2017-08-02 at 1 35 04 pm" src="https://user-images.githubusercontent.com/1472327/28887286-a116ec16-778a-11e7-94cc-454aa4cbeea2.png">
<img width="1440" alt="screen shot 2017-08-02 at 1 51 08 pm" src="https://user-images.githubusercontent.com/1472327/28887285-a1151774-778a-11e7-860f-196ea3b3e00d.png">

**Comment by krishongithub** [comment]
@soliton4 , I finally figured it out to compile a web socket client on Safari and receive packets from my Server, but I still have an issue here...

When I tried to play "mozilla_story.mp4" using mp4.js, it is working good and rendering. But when I try to play a video from h264 stream(as I mentioned above), the JS is receiving the frame properly, but it is unable to play the stream.. Is is something to do with the base profile you were talking about? Thanks.
